src/mqtt.cpp

```diff
@@ -0,0 +1,63 @@
+#include <PubSubClient.h>
+#include "secrets.h"
+#include "PluginManager.h"
+#include "mqtt.h"
+#include "plugins/WeatherPlugin.h"
+
+#ifdef ESP8266
+#include <ESP8266WiFi.h>
+#endif
+#ifdef ESP32
+#endif
+
+WiFiClient espClient;
+PubSubClient MQTTclient(espClient);
+
+void mqttCallback(char* topic, byte* payload, unsigned int length) {
+  Serial.print("Message arrived [");
+  Serial.print(topic);
+  Serial.print("] ");
+  String message;
+  for (int i = 0; i < length; i++) {
+    //Serial.print((char)payload[i]);
+    message += String((char)payload[i]);
+  }
+  if (topic == "Obegraensad/mode")
+  {
+    pluginManager.setActivePluginById(message.toInt());
+  } else {
+    pluginManager.getActivePlugin()->mqttHook(topic, message.toInt());
+  }
+  Serial.print(message);
+  Serial.println();
+}
+
+void mqttSetup(){
+    MQTTclient.setServer(MQTTSERVERIP, 1883);
+    MQTTclient.setCallback(mqttCallback);
+    mqttReconnect();
+}
+
+void mqttReconnect() {
+  // Loop until we're reconnected
+  while (!MQTTclient.connected()) {
+    Serial.print("Attempting MQTT connection...");
+    // Create a random client ID
+    String clientId = "ESPClient-";
+    clientId += String(random(0xffff), HEX);
+    // Attempt to connect
+    if (MQTTclient.connect(clientId.c_str())) {

```

Comment:
Wouild be nice if it works with username and password like:     if (MQTTclient.connect(clientId.c_str(), MQTT_USERNAME, MQTT_PASSWORD)) {

Reply:
Added as option. #define MQTT_AUTH to enable, then #define MQTT_USER and MQTT_PASS in secrets.